src/com/facebook/buck/worker/WorkerProcess.java

```diff
@@ -170,7 +217,85 @@ public synchronized void close() {
   }
 
   @VisibleForTesting
-  void setProtocol(WorkerProcessProtocol.CommandSender protocolMock) {
+  void launchForTesting(WorkerProcessProtocol.CommandSender protocolMock) {
     this.protocol = protocolMock;
+    handshakePerformed = true;
+    readerThread.start();
+  }
+
+  private void processNextCommandResponse() throws Throwable {
+    Preconditions.checkState(
+        protocol != null,
+        "Tried to submit a job to the worker process before the handshake was performed.");
+
+    Preconditions.checkState(
+        !Thread.holdsLock(this),
```

Comment:
This can't be false, why need we check it?

Reply:
I think as written, this is definitely true. This is mostly guarding to future changes in the code - if e.g. the caller is `synchronized` by accident.